### ISSUE TITLE
Bug in selfFiltering

### ISSUE THREAD

**Issue by janroz** [k-samuel/faceted-search#37]
Hello Kirill,

I thing I found a bug. 

When I have selfFiltering on, filters not returns counts for all possible params.

```
$filters[] = new ValueFilter('category', [1,4,5]);
$filters[] = new ValueIntersectionFilter('u', [10,12]);
$filters[] = new ValueFilter('mount', ['ABC', 'CDE']);

...

$query = (new AggregationQuery())->filters($filters)->countItems()->selfFiltering(true);
```

It works perfectly, it returns counts for all but not the categories that are not selected. Returns counts for category 1,4,5 but not others. When I select some other category, it returns counts for it, for example:

`$filters[] = new ValueFilter('category', [1,4,5,6,7]);`

it returns counts for 1,4,5,6,7.

I don't know if it is bug or feature. I tried to fix it, but without success. Can you please point me out?

**Comment by k-samuel** [comment]
Interesting. I’ll check it

**Comment by janroz** [comment]
I'll prepare some samples for you today. I need a self-filtering mechanism, as I mentioned earlier, because certain parameters need to work together using an AND operator, not OR. I need to list records that have both X and Y, not records having either X or Y.

I will try to create some examples. I can send you a link to a website where this is implemented, but only in a private message. If you're interested, please contact me.

**Comment by k-samuel** [comment]
Shure. Please write me  in FB (contacts on my profile)

**Comment by k-samuel** [comment]
Do I understand correctly that you would like to be able to enable self-filtering for specific filters and not for all ?

**Comment by janroz** [comment]
Exactly!

**Comment by k-samuel** [comment]
Simple implementation is ready. You can test it in your case. https://github.com/k-samuel/faceted-search/pull/38
I haven’t had time to cover it well with tests yet. If it’s conceptually suitable, I’ll try to add test cases.

**Comment by janroz** [comment]
Kirill, it looks very good!

**Comment by k-samuel** [comment]
Added a couple of test cases. Didn't reveal any unexpected behavior. Please let me know that in your case everything works as expected. If everything is ok, I’ll make a minor release.

**Comment by janroz** [comment]
Hi, it works perfectly. I took a closer look at it today. Great job!

**Comment by k-samuel** [comment]
Released in [3.2.1](https://github.com/k-samuel/faceted-search/releases/tag/3.2.1)
